scripts/check-precommit-reqs

```diff
@@ -11,23 +11,27 @@
 ##==============================================================================
 
 ## Check whether source code has been formatted (diff outputs '@@'):
-if ./scripts/format-code --whatif | grep -q "@@"; then
+if [[ $# -ne 0 ]]; then
+    files="$*"
+fi
+
+if ./scripts/format-code --whatif $([[ "$files" ]] && echo "--files=${files// /,}") | grep -q "@@"; then
     echo ""
     echo "Commit failed: please run ./scripts/format-code"
     echo ""
     exit 1
 fi
 
 # Check whether all sources have the license header:
-if ! ./scripts/check-license; then
+if ! ./scripts/check-license "$files"; then
```

Comment:
So, I was testing this, and when `$files` is empty, this is still sending the argument `""` (the empty string) to the `check-license` script, and so it jumps into the `$# -ne 0` logic. This doesn't seem right to me.